File: frontend/src/types/publishLog.ts
export type PublishLogStatus = "none" | "winner" | "loser" | "archived";

export const PUBLISH_LOG_STATUS_LABELS: Record<PublishLogStatus, string> = {
  none: "Chưa đánh giá",
  winner: "Winner",
  loser: "Loser",
  archived: "Đã lưu trữ",
};

// Kept here (not re-fetched from an AI Story module -- that generator was
// removed, see docs/features/63-remove-ai-content-and-insights.md) purely
// as manual labels for the free-text `story_style` field below.
export type StoryStyle =
  | "emotional"
  | "humorous"
  | "inspirational"
  | "dramatic"
  | "educational"
  | "sales";

export const STORY_STYLE_LABELS: Record<StoryStyle, string> = {
  emotional: "Cảm động",
  humorous: "Hài hước",
  inspirational: "Truyền cảm hứng",
  dramatic: "Kịch tính",
  educational: "Giáo dục",
  sales: "Bán hàng",
};

export interface PublishLog {
  id: number;
  video_id: number;
  video_title: string;
  video_topic: string | null;
  video_emotion: string | null;
  platform: string;
  page_name: string | null;
  hook_type: string | null;
  story_style: string | null;
  ai_story_job_id: number | null;
  affiliate_product: string | null;
  affiliate_clicks: number;
  affiliate_sales: number;
  affiliate_revenue: number;
  published_at: string;
  status: PublishLogStatus;
  notes: string | null;
  created_at: string;
}

export interface PublishLogCreateInput {
  video_id: number;
  platform?: string;
  page_name?: string;
  hook_type?: string;
  story_style?: string;
  ai_story_job_id?: number;
  affiliate_product?: string;
  affiliate_clicks?: number;
  affiliate_sales?: number;
  affiliate_revenue?: number;
  published_at?: string;
  status?: PublishLogStatus;
  notes?: string;
}

export interface PublishLogUpdateInput {
  page_name?: string;
  hook_type?: string;
  story_style?: string;
  affiliate_product?: string;
  affiliate_clicks?: number;
  affiliate_sales?: number;
  affiliate_revenue?: number;
  status?: PublishLogStatus;
  notes?: string;
}

File: backend/app/models/publish_log.py
from datetime import datetime, timezone

from sqlalchemy import DateTime, Float, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base
from app.models.video import Video

PUBLISH_LOG_STATUSES = ("none", "winner", "loser", "archived")


def _utcnow() -> datetime:
    return datetime.now(timezone.utc)


class PublishLog(Base):
    """Records that a Library Video was published somewhere, plus the
    creative/business metadata that can never be extracted from a platform's
    own analytics export (hook_type, story_style, ai_story_job_id,
    affiliate_*) -- see docs/features/15-performance-intelligence.md.

    `ai_story_job_id`/`story_style` are free-form traceability fields the
    user can fill in manually; the AI Story generator module and the
    Insights CSV-import feature that used to populate/link these were
    removed (see docs/features/63-remove-ai-content-and-insights.md).
    `post_id`/`page_id` are dormant leftovers from that removed Insights
    linking feature -- kept on the model (SQLite has no migration
    framework to drop them) but no longer set or read by any code.
    """

    __tablename__ = "publish_log"

    id: Mapped[int] = mapped_column(primary_key=True)
    video_id: Mapped[int] = mapped_column(ForeignKey("video.id"), nullable=False)

    platform: Mapped[str] = mapped_column(String, nullable=False, default="facebook")
    page_name: Mapped[str | None] = mapped_column(String, nullable=True)

    hook_type: Mapped[str | None] = mapped_column(String, nullable=True)
    story_style: Mapped[str | None] = mapped_column(String, nullable=True)
    ai_story_job_id: Mapped[int | None] = mapped_column(Integer, nullable=True)

    affiliate_product: Mapped[str | None] = mapped_column(String, nullable=True)
    affiliate_clicks: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    affiliate_sales: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    affiliate_revenue: Mapped[float] = mapped_column(Float, nullable=False, default=0)

    published_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=_utcnow)
    status: Mapped[str] = mapped_column(String, nullable=False, default="none")

    post_id: Mapped[str | None] = mapped_column(String, nullable=True)
    page_id: Mapped[str | None] = mapped_column(String, nullable=True)

    notes: Mapped[str | None] = mapped_column(String, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=_utcnow)

    video: Mapped[Video] = relationship("Video", lazy="joined")
